server/app_engine.py

```diff
@@ -0,0 +1,1304 @@
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__author__ = "lizlooney@google.com (Liz Looney)"
+
+# Python Standard Library
+from datetime import datetime, timedelta, timezone
+from functools import wraps
+import json
+import logging
+import time
+
+# Other Modules
+import flask
+from flask import Flask
+from flask_oidc_ext import OpenIDConnect
+from sqlitedict import SqliteDict
+import sentry_sdk
+from sentry_sdk.integrations.flask import FlaskIntegration
+
+# My Modules
+import action
+import bbox_writer
+import blob_storage
+import cloud_secrets
+import constants
+from credentialstore import CredentialStore
+import dataset_producer
+import dataset_zipper
+import exceptions
+import frame_extractor
+import model_trainer
+import roles
+from roles import Role
+import storage
+import team_info
+import tflite_creator
+import tracking
+import util
+
+
+
+sentry_dsn = cloud_secrets.get_or_none('sentry_dsn')
+if sentry_dsn is not None:
+    sentry_sdk.init(
+        dsn=sentry_dsn,
+        integrations=[FlaskIntegration()],
+
+        # Set traces_sample_rate to 1.0 to capture 100%
+        # of transactions for performance monitoring.
+        # We recommend adjusting this value in production.
+        traces_sample_rate=1.0,
+
+        # By default the SDK will try to use the SENTRY_RELEASE
+        # environment variable, or infer a git commit
+        # SHA as release, however you may want to set
+        # something more human-readable.
+        # release="myapp@1.0.0",
+        )
+
+app = flask.Flask(__name__)
+
+app.config.update(
+    {
+        # Flask properties
+        "SECRET_KEY": cloud_secrets.get("flask_secret_key"),
+        "MAX_CONTENT_LENGTH": 8 * 1024 * 1024,
+        "ALLOWED_EXTENSIONS": {'png', 'jpg', 'jpeg', 'gif'},
+
+        # OIDC properties
+
+        "OIDC_ID_TOKEN_COOKIE_SECURE": False,
+        "OIDC_REQUIRE_VERIFIED_EMAIL": False,
+        "OIDC_SCOPES": ["openid", "email", "roles"],
+    }
+)
+
+if util.is_development_env():
+    app.debug = True
+    app.testing = True
+else:
+    app.debug = False
+    app.testing = False
+
+
+#
+# If a redis server is specified, use it, otherwise use a
+# local sqlite database.
+#
+payload = cloud_secrets.get_or_none("client_secrets")
+if payload is not None:
+    credentials_dict = json.loads(payload)
+    app.config.update({"OIDC_CLIENT_SECRETS": credentials_dict})
+    if constants.REDIS_IP_ADDR is not None:
+        oidc = OpenIDConnect(app, credentials_store=CredentialStore())
+    else:
+        oidc = OpenIDConnect(app, credentials_store=SqliteDict('users.db', autocommit=True))
+else:
+    oidc = None
+
```

Comment:
Can we perhaps keep the use_oidc boolean?  It doesn't have to be in constants, but a) I like it better because it expresses intent in a human readable fashion, and b) if perchance OpenIDConnect() returns None, then we've effectively turned off oidc which is expressly not the intent and might mask a problem if the project happened to have a team info file.

Reply:
Sure. Since it's only used in app_engine.py, I'll put it there.